src/main/java/Deadline.java

```diff
@@ -0,0 +1,50 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+// partial solution copied from iP
+public class Deadline extends Task {
+
+    protected LocalDate date;
+    protected LocalTime time;
+
+    /**
+     * Executes the deadline command, causing add a task of type Deadline to the taskList,
+     * provided that the command input is valid.
+     *
+     * @param description String describing the Deadline task.
+     * @param by String indicating the deadline of the Deadline task.
+     * @param isDone Boolean indicating if the Deadline task is done.
+
+     */
+    public Deadline(String description, String by, boolean isDone) {
```

Comment:
Perhaps you can change the name "by" to something that is easier to understand? For instance deadlineDate or ddlDate or just date?